meeteval/wer/matching/greedy_combination_matching.py

```diff
@@ -0,0 +1,216 @@
+import itertools
+import functools
+from typing import List, Iterable
+
+
+import numpy as np
+
+from meeteval.io.seglst import SegLST
+from meeteval.wer.matching.cy_greedy_combination_matching import cy_forward_col
+
+
+def _apply_assignment(assignment, segments, n=None):
+    if n is None:
+        n = max(assignment) + 1
+    else:
+        n = max(max(assignment) + 1, n)
+        assert n > max(assignment), (n, max(assignment))
+    h_ = [[] for _ in range(n)]
+    for a, h in zip(assignment, segments):
+        h_[a].append(h)
+    return h_
+
+
+def _greedy_correct_assignment(
+        segments: List[np.ndarray],
+        streams: List[np.ndarray],
+        assignment: List[int],
+        forward_column: callable = cy_forward_col,
+):
+    """
+    Greedily modify the assignment (of the segments to the streams)
+    to decrease the Levenshtein distance.
+
+    Args:
+        segments: A list of segments for which stream labels should be obtained
+        streams: A list of streams to which the segments are assigned
+        assignment: The initial assignment.
+        forward_column: A function to compute a column update. Can be used
+            to switch between different cost functions, e.g. with different
+            substitution costs or with or without time constraint.
+    """
+    # Temporary variables
+    cost_differences = np.zeros((len(streams),), dtype=int)
+    costs_with_utterance = np.zeros((len(streams),), dtype=int)
+    costs_without_utterance = np.zeros((len(streams),), dtype=int)
+    updated_forward_columns = [None] * len(streams)
+
+    # Iteratively update assignment greedy
+    for num_updates in range(1, 50):
```

Comment:
Add a comment, that 50 is chosen to be far too large?
If I remember correctly, num_updates didn't go larger than 3 or 4.

Reply:
Done